pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -99,6 +99,25 @@ func createLogicalReplicationStreamPlanHook(
 			return pgerror.New(pgcode.InvalidParameterValue, "the same number of source and destination tables must be specified")
 		}
 
+		options, err := evalLogicalReplicationOptions(ctx, stmt.Options, exprEval, p)
+		if err != nil {
+			return err
+		}
+
+		mode := jobspb.LogicalReplicationDetails_Immediate
+		if m, ok := options.GetMode(); ok {
+			switch m {
+			case "validated":
+				mode = jobspb.LogicalReplicationDetails_Validated
+			case "immediate":
+			default:
+				return pgerror.Newf(pgcode.InvalidParameterValue, "unknown mode %q", m)
+			}
+		}
+		if len(options.userFunctions) > 0 || options.defaultFunction != nil && options.defaultFunction.FunctionId != 0 {
+			mode = jobspb.LogicalReplicationDetails_Validated
+		}
```

Comment:
If the user `specifies` immediate but has a UDF, should that be an error?

Reply:
Done.